src/api/cp.py

```diff
@@ -42,3 +42,48 @@ def get_all_cp_machines() -> List[Dict[str, Any]]:
     except ValueError as json_err:
         logging.error(f"JSON decoding error: {json_err}")
         raise json_err
+
+
+def get_computing_providers_list(region: str) -> List[Dict[str, Any]]:
+    """
+    Fetches a list of computing providers from the API based on the given region and hardware.
+
+    Args:
+        region (str): The region for which computing providers are requested.
+
+    Returns:
+        List[Dict[str, Any]]: A list of dictionaries containing details of computing providers.
+
+    Raises:
+        SwanHTTPError: If the API call fails.
+        SwanConnectionError: If there is a network problem connecting to the API.
+        SwanTimeoutError: If the request to the API times out.
+        ValueError: If the response from the API cannot be decoded.
+        Exception: For other types of exceptions.
+
+    """
+    if not region:
+        raise ValueError("Please provide a valid Region")
+
+    url = f"{SWAN_API}{CP_LIST}"
+    data = {"region": region}
+
+    try:
+        response = requests.post(url, data=data)
```

Comment:
please add authorization

Reply:
merge all of the PRs, Ill add ApiClient on them